docs/UpgradeGuide.md

```diff
@@ -2,57 +2,78 @@
 
 # From 0.x to 1.0
 
-- [Namespacing & the public API](#namespacing--the-public-api)
-- [Configuration](#configuration)
-- [Instrumentation](#instrumentation)
-- [Full list of breaking changes](#full-list-of-breaking-changes)
+Upgrading `ddtrace` from 0.x to 1.x introduces some breaking changes which are outlined below.
 
-Upgrading `ddtrace` from 0.x to 1.x introduces some changes to namespacing, the public API, and the underlying trace data structure.
+**How to upgrade basic usage**
 
-Here's a list of the most common changes you may encounter:
+For users with a basic implementation (configuration file + out-of-the-box instrumentation), only minor changes to your configuration file are required: most applications take just minutes to update. Check out the following sections for a step-by-step guide.
 
-## Namespacing & the public API
+- [Configuration](#1.0-configuration)
+  - [`require` paths have changed](#1.0-configuration-requires)
+  - [Settings have been renamed](#1.0-configuration-settings)
+  - [Activating instrumentation](#1.0-configuration-instrumentation)
+- [Instrumentation](#1.0-instrumentation)
+  - [Service naming](#1.0-instrumentation-service-naming)
 
-To avoid naming conflicts with new Datadog features and products, many of the tracing constants and functions defined directly under `Datadog` have moved to `Datadog::Tracing`.
+**Addtional upgrades for advanced usage**
 
-You can access most common functions through our [public API](https://www.rubydoc.info/gems/ddtrace/) within `Datadog::Tracing`. Use of some of the functions in the public API will be described in use cases below. Here's a few examples:
+For users with an advanced implementation of `ddtrace` (custom instrumentation, sampling or processing behavior, etc), some namespace and behavioral changes may be required. See the following sections for details about what features changed and how to use them.
 
```

Comment:
I guess it makes sense to underline the "additionalness" of these steps again here:

```suggestion
For users with an advanced implementation of `ddtrace` (custom instrumentation, sampling or processing behavior, etc), some namespace and behavioral changes may be additionally required. On top of the changes listed for base usage, see the following sections for details about what features changed and how to use them.
```